test/integration/replicaset/replicaset_test.go

```diff
@@ -573,23 +557,20 @@ func TestSpecReplicasChange(t *testing.T) {
 	// Add a template annotation change to test RS's status does update
 	// without .Spec.Replicas change
 	rsClient := c.Extensions().ReplicaSets(ns.Name)
-	updateRS(t, rsClient, rs, func(rs *v1beta1.ReplicaSet) {
+	newRS := updateRS(t, rsClient, rs.Name, func(rs *v1beta1.ReplicaSet) {
 		rs.Spec.Template.Annotations = map[string]string{"test": "annotation"}
 	})
-	newRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
-	if err != nil {
-		t.Fatalf("failed to update rs %s: %v", rs.Name, err)
-	}
-	if newRS.Generation <= rs.Generation {
-		t.Fatalf("Failed to verify .Generation has incremented for rs %s: %v", rs.Name, err)
+	savedGeneration := newRS.Generation
+	if savedGeneration == rs.Generation {
```

Comment:
There is not need to check whether `savedGeneration` is less than `rs.Generation` as `Generation` can only be incremented.